bravado/mapping/model.py

```diff
@@ -0,0 +1,147 @@
+    cases for interactive debugging in a REPL.
+
+    :param model_name: model name
+    :param model_dict: json-like dict that describes a model.
+    :returns: dynamic type created with attributes, docstrings attached
+    :rtype: type
+    """
+    props = model_dict['properties']
+
+    methods = dict(
+        __doc__=docstring_property(partial(create_model_docstring, props)),
+        __eq__=lambda self, other: compare(self, other),
+        __init__=lambda self, **kwargs: set_props(self, **kwargs),
+        __repr__=lambda self: create_model_repr(self),
+        __dir__=lambda self: props.keys(),
+        _flat_dict=lambda self: create_flat_dict(self),
+        _swagger_types=swagger_type.get_swagger_types(props),
+        _required=model_dict.get('required'),
+    )
+    return type(str(model_name), (object,), methods)
+
+
+def compare(first, second):
+    """Compares two model types for equivalence.
+
+    TODO: If a type composes another model type, .__dict__ recurse on those
+          and compare again on those dict values.
+
+    :param first: generated model type
+    :type first: type
+    :param second: generated model type
+    :type second: type
+    :returns: True if equivalent, False otherwise
+    """
+    if not hasattr(first, '__dict__') or not hasattr(second, '__dict__'):
+        return False
+
+    # Ignore any '_raw' keys
+    def norm_dict(d):
+        return dict((k, d[k]) for k in d if k != '_raw')
+
+    return norm_dict(first.__dict__) == norm_dict(second.__dict__)
+
+
+def set_props(model, **kwargs):
+    """Constructor for the generated type - assigns given or default values
+
+    :param model: generated model type
+    :type model: type
+    :param kwargs: attributes to override default values of constructor
+    :type kwargs: dict
+    """
+    types = getattr(model, '_swagger_types')
+    arg_keys = kwargs.keys()
+    for prop_name, prop_swagger_type in types.iteritems():
+        python_type = swagger_type.swagger_to_py_type(prop_swagger_type)
+        # Assign all property values specified in kwargs
+        if prop_name in arg_keys:
+            prop_value = kwargs[prop_name]
+            arg_keys.remove(prop_name)
+        else:
+            # If not in kwargs, provide a default value to the type
+            prop_value = swagger_type.get_instance(python_type)
+        setattr(model, prop_name, prop_value)
+    if arg_keys:
+        raise AttributeError(" %s are not defined for %s." % (arg_keys, model))
+
+
+def create_model_repr(model):
+    """Generates the repr string for the model.
+
+    :param model: generated model type
+    :type model: type
+    :returns: repr string for the model
+    """
+    string = ""
+    separator = ""
+    for prop in sorted(getattr(model, '_swagger_types').keys()):
+        string += ("%s%s=%r" % (separator, prop, getattr(model, prop)))
+        separator = ", "
+    return "%s(%s)" % (model.__class__.__name__, string)
```

Comment:
This is the wrong way to do string building, I guess it's just copied, but this should be 

```python

repr = [
    ("%s%s=%r" % (separator, prop, getattr(model, prop)))
   for prop in sorted(getattr(model, '_swagger_types'))
]
return return "%s(%s)" % (model.__class__.__name__, repr.join(", "))
```